### ISSUE TITLE
serve_dir not working as expected

### ISSUE THREAD

**Issue by xorxsan** [http-rs/tide#774]
Hi!

   I've an issue with `serve_dir()`. When filenames inside the folder I want to serve begin with the same name as the path specified in the `at()` method, the prefix is stripped and the request throws a 404. This is what I'm doing:
```rust
#[async_std::main]
async fn main() -> Result<(), std::io::Error> {
    tide::log::start();
    let mut app = tide::new();
    app.at("/swagger").serve_dir("./swagger/")?;
    app.at("/texts").serve_dir("./texts/")?;
    app.listen("0.0.0.0:5000").await?;
    Ok(())
}
```
In the `swagger` folder there are a bunch of files that start with `swagger` (i.e.: swagger-ui.css) and in the folder `texts` I've added an index.html and a file called texts1.txt. When I make the following requests:
```
http://localhost:5000/swagger/swagger-ui.css
http://localhost:5000/texts/texts1.txt
```

I always get 404 and these are the logs:
```
tide::log::middleware <-- Request received
    method GET
    path /swagger/swagger-ui.css
tide::fs::serve_dir Requested file: "\\\\?\\C:\\Users\\xxxxxxxxx\\yyyy\\zzzzzzz\\my-tests\\swagger\\-ui.css"
tide::fs::serve_dir File not found: PathBuf { inner: "\\\\?\\C:\\Users\\xxxxxxxxx\\yyyy\\zzzzzzz\\my-tests\\swagger\\-ui.css" }
tide::log::middleware Client error --> Response sent
    method GET
    path /swagger/swagger-ui.css
    status 404 - Not Found
    duration 9.0704ms
```
However he request `http://localhost:5000/texts/index.html` works as expected.

The workaround would be to rename the path to another thing so it does not match any of the files inside the folder. But still...
Am I doing something wrong?

Tide version: 0.15.0
Rust compiler: rustc 1.48.0 (7eac88abb 2020-11-16)
OS: Windows 10 x64

Thanks!

**Comment by mendelt** [comment]
Hi, sorry for the late reply and thank you for notifying us. I wrote a small [example](https://github.com/mendelt/tide_serve_dir_bug) based on your code to reproduce this.
I could reproduce the issue with Tide 0.15.1 but it seems to be fixed in Tide 0.16.0. Could you confirm this?
Thank you!

**Comment by xorxsan** [comment]
Hi @mendelt

   I tried with 0.16.0 as you suggested and it's working. I'll close this ticket.
 
   Thanks!
